src/main/kotlin/com/swisscom/health/des/cdr/clientvm/handler/FileHandlingBase.kt

```diff
@@ -61,16 +73,33 @@ abstract class FileHandlingBase(protected val cdrClientConfig: CdrClientConfig,
     }
 
     /**
-     * Build headers with connector-id, function key, processing mode and trace id.
+     * Build headers with connector-id, access token, processing mode and trace id.
      */
     protected fun buildBaseHeaders(connectorId: String, mode: CdrClientConfig.Mode): Headers {
         val traceId = tracer.currentSpan()?.context()?.traceId() ?: ""
-        return Headers.Builder().run {
-            this[CONNECTOR_ID_HEADER] = connectorId
-            this[FUNCTION_KEY_HEADER] = cdrClientConfig.functionKey
-            this[CDR_PROCESSING_MODE_HEADER] = mode.value
-            this[AZURE_TRACE_ID_HEADER] = traceId
-            this.build()
+        // TODO: Remove this check once the token is required
+        val accessToken = if(cdrClientConfig.idpCredentials.tenantId != DEFAULT_TENANT_ID) {
```

Comment:
You could return the token string from `getAccessToken()`, use an empty string in the `else` case and then test for `isNotBlank` to decide whether you should add the bearer token authZ header?

Reply:
But that would mean that we called the token service with some wrong data. If we already know that we don't want to call it then I'd rather not do it.